packages/swirl-components/src/components/swirl-table/swirl-table.tsx

```diff
@@ -228,10 +237,49 @@ export class SwirlTable {
 
       const columnWidth =
         column.width || `${column.getBoundingClientRect().width}px`;
+      const isLastColumnSticky =
+        column.sticky && columns.length === colIndex + 1;
+      const hasShadowRight =
+        column.sticky &&
+        !columns.slice(colIndex + 1, columns.length - 1).some((c) => c.sticky);
 
       cellsOfColumn.forEach((cell) => {
```

Comment:
this whole function is quite big. i would break it out a little. 

I think a callback function that is a private method of this class would be the first step. additionally you *could* split out the functionality to handle the cell stickiness and the column stickiness. 

This has the advantage of future debugging to have the logic seperated based on the behaviour on a certain element. e.g. `handleCellStyl` and `handleColumnStyle` in a orchestrating `handleStyle` function. 

names should emerge when you start breaking out the function :)

Reply:
I agree, thanks for the suggestion :) I broke down the function into various smaller functions. Please have a look